src/toil/job.py

```diff
@@ -2450,35 +2457,31 @@ class PromisedRequirementFunctionWrappingJob(FunctionWrappingJob):
     """
     Handles dynamic resource allocation using :class:`toil.job.Promise` instances.
     Spawns child function using parent function parameters and fulfilled promised
-    resource requirements. See :class:`toil.job.FunctionWrappingJob` class
+    resource requirements. This job must be encapsulated (see :func:`toil.job.encapsulate`)
+    to avoid a race condition between child functions.
```

Comment:
I feel like this explanation needs more meat.

Also it is an implementation detail that should not be mentioned in the docstring. Otherwise you are going to confuse our users with something that is irrelevant to them.